docs/source/conf.py

```diff
@@ -243,9 +276,33 @@ def process_modin_accessors(args):
 
 
 def setup(app):
+    # Make sure modin.pandas namespace is properly set up
+    import modin.pandas as pd
+    import snowflake.snowpark.modin.plugin
+    # Monkeypatch dt/str to make sure their children are resolvable by autosummary-generate.
+    # Without this monkeypatch, the autosummary-generate (which runs before any custom documenter
+    # classes can take effect) will report an error like the following for every child of Series.str
+    # and Series.dt:
+    #
+    # WARNING: [autosummary] failed to import modin.pandas.Series.str.slice.
+    # Possible hints:
+    # * AttributeError: 'property' object has no attribute 'slice'
+    # * ImportError: 
+    # * ModuleNotFoundError: No module named 'modin.pandas.Series'
+    #
+    # Because we're replacing the `property` object, we also need to set the __doc__ of the new
```

Comment:
what do we mean here by "replacing the `property` object"

Reply:
Originally, `Series.str` is a python `property`, which has its __doc__ attribute set to the expected docstring. We're replacing it with the `StringMethods`/`DatetimeProperties` class instead, which has a different docstring.